video/out/gpu/video.c

```diff
@@ -1702,20 +1702,21 @@ static void reinit_scaler(struct gl_video *p, struct scaler *scaler,
                           double scale_factor,
                           int sizes[])
 {
+    assert(conf);
```

Comment:
Not sure about this assert? Can't conf be empty sometimes like if no scale is explicitly configured.

Reply:
We dereference it on next line already. Do you see better way of handling it?

Reply:
Oh yeah. I guess maybe I was thinking of that time when there was a segfault with no dscale, but I guess things changed since then.

Reply:
If it was, it would be crashing right now due to `reinit_scaler` always trying to de-reference the pointer as the second argument to `scaler_conf_eq`, I think?

Quickly checked the call sites of it through `pass_sample(` and this seems to be the case that it should be set as it is getting de-referenced a bunch in the various functions such as `pass_read_video`.